examples/basic/evm-example/src/components/SwapContent/SwapContent.tsx

```diff
@@ -0,0 +1,599 @@
+      const allBalances = (await Promise.allSettled(promises)).flatMap((p) =>
+        p.status === 'fulfilled' ? p.value.wallets : []
+      )
+      setCosmosBalances(allBalances)
+    }
+
+    if (keplrWalletState.connected && keplrWalletState.accounts) {
+      const promises =
+        keplrWalletState.accounts.map((account) =>
+          sdk.balance({
+            address: account?.split(':')?.[1] || '',
+            blockchain: account?.split(':')?.[0] || '',
+          })
+        ) || []
+
+      getBalances(promises)
+    }
+  }, [keplrWalletState.connected, keplrWalletState.accounts])
+
+  useEffect(() => {
+    setFromToken(null)
+  }, [fromChain])
+
+  useEffect(() => {
+    setToToken(null)
+  }, [toChain])
+
+  const handleSwitchFromAndTo = () => {
+    setFromChain(toChain)
+    setFromToken(toToken)
+    setToChain(fromChain)
+    setToToken(fromToken)
+  }
+
+  const getWalletByChainType: (chainType: string) => WalletTypes = (
+    chainType
+  ) => {
+    if (chainType === TransactionType.EVM) {
+      return WalletTypes.META_MASK
+    } else if (chainType === TransactionType.SOLANA) {
+      return WalletTypes.PHANTOM
+    } else {
+      return WalletTypes.KEPLR
+    }
+  }
+
+  const getAddressByChain = (chain: BlockchainMeta | null) => {
+    if (!chain) return null
+
+    console.log(chain)
+
+    if (chain.type === TransactionType.EVM) {
+      return metamaskWalletState.accounts?.[0]?.split(':')?.[1]
+    } else if (chain.type === TransactionType.SOLANA) {
+      return phantomWalletState.accounts?.[0]?.split(':')?.[1]
+    } else if (chain.type === TransactionType.COSMOS) {
+      return keplrWalletState.accounts
+        ?.find((account) => account.split(':')?.[0] === chain.name)
+        ?.split(':')?.[1]
+    }
+  }
+
+  const checkConnectedWalletError = async () => {
+    if (
+      (fromChain?.type === TransactionType.EVM ||
+        toChain?.type === TransactionType.EVM) &&
+      !metamaskWalletState.connected
+    ) {
+      try {
+        await connect(WalletTypes.META_MASK)
+      } catch (error) {
+        console.log(error)
+
+        return 'Error connecting to Metamask. Please check Metamask and try again.'
+      }
+    }
+
+    if (
+      (fromChain?.type === TransactionType.COSMOS ||
+        toChain?.type === TransactionType.COSMOS) &&
+      !keplrWalletState.connected
+    ) {
+      try {
+        await connect(WalletTypes.KEPLR)
+      } catch (error) {
+        console.log(error)
+
+        return 'Error connecting to Keplr. Please check Keplr and try again.'
+      }
+    }
+
+    if (
+      (fromChain?.type === TransactionType.SOLANA ||
+        toChain?.type === TransactionType.SOLANA) &&
+      !phantomWalletState.connected
+    ) {
+      try {
+        await connect(WalletTypes.PHANTOM)
+      } catch (error) {
+        console.log(error)
+
+        return 'Error connecting to Phantom. Please check Phantom and try again.'
+      }
+    }
+
+    return false
+  }
```

Comment:
You could refactor this section to avoid duplicate codes.

Reply:
done